### DISCUSSION TITLE
CB3S 4-433 Rohs

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16027] [Q&A] [answered]
I've been searching the net but cannot confirm if this will run on this device.
https://www.amazon.com/gp/product/B09M6HTDVS/
It has a CB3S 4-433 Rohs chip which doesn't seem to be supported by tasmota. 

Is this correct? If so, I'll send it back for another that is supported.

**Comment by s-hadinger** [comment, marked as answer]
No, it's not based on Espressif chip.

**Reply by dac2020** [reply]
Thank you for the quick reply. 
I'll send it back for this one as some seem to be saying they are using it with tasmota.

https://www.amazon.com/gp/product/B071KFX63R/

**Reply by sfromis** [reply]
Some may have been using a prior batch which had a Tasmota-compatible chip. You *could* also replace the CB3S module with a supported ESP-12 module, if you like.

**Reply by dac2020** [reply]
I would not be able to do that kind of work but now I'm not sure what I should order. 
Maybe there is a list of supported hardware somewhere?
Do you know of a compatible 4 channel relay with case? I could order some of those knowing they would work.

**Reply by sfromis** [reply]
Supported hardware: https://templates.blakadder.com/
Devices listed there were compatible at the time of inclusion, but there is a risk that designs are no longer the same.

**Reply by barbudor** [reply]
From AliEx, I would search for "ESP32 4 relays" which gives a few results
ESP32 is a more capable CPU which would some crappy designs based on ESP8266 + additional MCU
The Robodyn model looks nice with no compromises

**Reply by dac2020** [reply]
Thank you! I see this on Amazon and could have it by Sat.
https://www.amazon.com/LILYGO-T-Relay-Wireless-Development-Control/dp/B09XX1HL9Z/

Problem is, I don't see any leads for serial and don't have the kind of soldering iron to do this.

**Reply by barbudor** [reply]
Take the tu2kit version. Comes with their flashing adapter as the board is using an false usb-c connector

**Reply by dac2020** [reply]
Done, ordered. 

I see a Shelly device in the hardware list but there isn't one by that name on Amazon.
The closest I can see is https://smile.amazon.com/gp/product/B07Z623J8Z/

Does anyone know if this is the same as this one;
https://templates.blakadder.com/shelly_plus_2PM.html

Thank you so much for the help. This is my introduction to tasmota to learn a little about it.

**Reply by sfromis** [reply]
The Shelly 2.5 is an older version. Similar function with two relays and energy monitoring. The "Plus" version is an updated design including ESP32 instead of ESP8266. While the better MCU does not matter much for basic Tasmota operation, it does provide extra features. Both can be flashed without opening, connecting wires. IIRC, the possibility for OTA upgrade of the Shelly 2.5 to Tasmota is not available in the Plus design.

**Reply by dac2020** [reply]
That's great, I'll order one of those as well to see how I might use it for energy monitoring.

**Reply by dac2020** [reply]
I'll search around and see what I can find. 
I'd like something tiny like the Shelly to monitor the current of four solar strings of 2.5KW each. 
I'll use the relay board I just bought to control a sprinkler system.

These two things would be perfect projects to start learning about tasmota.

**Reply by dac2020** [reply]
> Take the tu2kit version. Comes with their flashing adapter as the board is using an false usb-c connector

Got it just now. It didn't come with anything but the board. 
I'll search for some sort of documentation that explains what I need to do now with my first hands on experience with tasmota.

**Reply by sfromis** [reply]
Without the U2T kit, the board is not terribly friendly to flashing, as the pins needed are not broken out elsewhere. The most practical with what you have is probably to find the relevant pins on the ESP32-WROVER module, and use a USB/serial adapter.
https://tasmota.github.io/docs/Pinouts/#esp32-wrover

**Reply by sfromis** [reply]
Much less hassle if you had got this one: 
![image](https://user-images.githubusercontent.com/47082390/180581107-4d7489d5-0ed3-45cf-a73a-5148521dec0e.png)

**Reply by dac2020** [reply]
Yes, I didn't look very closely and thought the suggestion above meant it's included. I can order that and wait some more but I can also try the serial method since a header came with it. 

Thanks for the info on the board too. Quick look seems to show pins 17 and 18 are what I'll need. Maybe I'll give it a try. Hate to ruin the board if I mess it up though.

**Reply by dac2020** [reply]
For the life of me, I cannot find any information on what pins I need to connect to get serial access.
The image in the link you shared shows gpio1 tx, gpio3 rx and gpio9 rx, gpio10 tx.
However, the board doesn't have any of those, it only shows IOxx pins and none are the above. 
It could be pins IO34 and IO35 so I tried that but I get nothing.

Wish I'd have known about the t-u2t but am trying to learn this while being overworked on too many other things at the same time.

Does anyone know what the rx/tx pins are on this device?

**Reply by barbudor** [reply]
Pinout of the ESP32 module from which some are available on the header:
https://tasmota.github.io/docs/Pinouts/#esp32-wrover

Rx/TX will be available only on the ESP32 module

**Reply by sfromis** [reply]
You *could* also "sacrifice" an USB-C cable, and connect the wires to a USB/serial adapter, after working out what wires have what function. That would basically recreate the dongle from Lilygo.

**Reply by dac2020** [reply]
Yes,, I have the esp32 module on this. ESP32-WROVER-B but as mentioned, that link being shared, I don't have those pins in the header. 
The header labels are all IOxx and none of the numbers are pins I've found anywhere in this link or on the net.
The top pins on the right show 34 and 35 and I have IO34 and IO35 on the header label but those didn't work.

I have C-cables I can sacrifice for sure. Just not sure I can deal with such tiny wires. The main problem would be 'working out' which wires are which.

![fef0f504-7f55-43af-9980-79dcb55cbdc6 __CR0,0,970,600_PT0_SX970_V1___](https://user-images.githubusercontent.com/9727841/180611053-8fa013ab-5d4d-4932-b0f9-117c26bd67fa.png)

These is this image in the sales info on Amazon with this board but even that doesn't show what those pins go to.

**Reply by dac2020** [reply]
I tore the thickest cable I have apart and there are only 4 wires and two are obviously power so, let's see where this gets me. I wonder if I should stop posting on this topic and create a new one for others using this board.

**Reply by sfromis** [reply]
In that case, it is not a "full" USB-C cable, as the connector has 2x12 pins. Normally, that means 12 connections, as as the plug can be inserted either way, but technically you could have it act differently depending on the orientation. With newer USB standards, you have 20+ wires end to end, as there is no reason not to use all pins just because you can twist the connector.

**Reply by dac2020** [reply]
Correct, it's not a full cable but I hoped it meant it would simply have power and data connectors but so far, I cannot get much from a serial connection, just a lot of garbage at all baud rates and flow control settings. Definitely what seems to be a stream of data as it corresponds with the initial booting and testing of each relay then a pulse.

**Reply by dac2020** [reply]
I started a new thread since this one was mainly asking about compatible hardware while now I'm trying to work with this specific hardware so probably best to have its own post to help someone else.

https://github.com/arendst/Tasmota/discussions/16054

**Reply by dac2020** [reply]
Got it. See other thread please.

**Reply by dac2020** [reply]
Ok, got the new board today and this time, with the adapter :).
https://www.amazon.com/dp/B09XWWYK4V

Sure don't want to damage this one so if anyone can share a link on how to go about flashing this, it would be most appreciated so I can finally try tasmota.

**Reply by sfromis** [reply]
I've not used the dongle, but assuming that it does what it says, just standard flashing like https://tasmota.github.io/install/

**Reply by dac2020** [reply]
I thought the adapter that comes with the board needs to have the firmware copied onto it then connected to the board as I start it. I cannot find enough information to know how to do this.

**Reply by sfromis** [reply]
I just gave you the link for how to flash Tasmota on to the board when you have USB connection. Of course, without having tried it, I can't be 100%, but I'd be surprised if it would not work. It should auto-detect chip type, meaning that you do not need to select which binary to flash.

**Reply by dac2020** [reply]
Yes, I was at that page. It says to try chrome since the browser is not supported. I'm running KDE so I'll have to install chrome I guess and see if Install shows up.

**Reply by sfromis** [reply]
The Tasmota Web Installer is not the only option, just the recommended one. Also see https://tasmota.github.io/docs/ESP32/#flashing

**Reply by dac2020** [reply]
LOL, my luck is just sucking with this.
I'm using KDE so can only install Chromium. 
When I connect the device, I can see the new USB/serial port but the installer never does.

https://tasmota.github.io/install/
Failed to open serial port.

Likely unable to reach the serial port from KDE... permissions thing or something along those lines.
I'll keep trying.

**Reply by dac2020** [reply]
Yes, I changed the device permissions and that worked.
sudo chmod 666 /dev/ttyACM0

Got it flashed and can finally access a tasmota device. 

Thanks so much everyone for the help. I'll play with this for a while, try to learn what I can and go from there.

**Reply by chumaty** [reply]
@dac2020 
hi, can you tell me how you flashed this device?  Which pins did you connect the wires to and how did you enter the chip into firmware mode?

**Comment by Vincent835** [comment]
> I've been searching the net but cannot confirm if this will run on this device. https://www.amazon.com/gp/product/B09M6HTDVS/ It has a CB3S 4-433 Rohs chip which doesn't seem to be supported by tasmota.
> 
> Is this correct? If so, I'll send it back for another that is supported.

I saw an alternative for this chip
https://github.com/openshwprojects/OpenBK7231T_App

**Reply by panciunio** [reply]
The CB3S was not mentioned there.
Still waiting for support... my favorite Sonoff S-55 branded socket changed ESP8288 to CB3S and it's unusable now...

**Reply by Jason2866** [reply]
@panciunio no support for either

**Comment by starbrights** [comment]
I think that one should help:
https://www.elektroda.de/rtvforum/topic3983019.html

I am completly new to that topic and found my tuya device includes a CB3S (BK7231N). I hope I will understand what is going on there. If someone can offer a little help in case ... I would be happy.

**Reply by Jason2866** [reply]
@starbrights Not related to Tasmota. So here is the wrong place to ask for help.

**Reply by barbudor** [reply]
Simple: CB3S = No Tasmota

If you want Tasmota: change the chip https://blakadder.com/replace-tuya-esp12/

If you don't want to change the chip, try OpenBK as listed above

If you want neither, throw it to the bin

**Reply by starbrights** [reply]
You are right. Big sorry! New to this topic. I want to get rid of Tuya, wether this is done by Tasmota or not. Changing to ESP is to hard for me. I hope I can add OpenBK somehow to openHAB finally.

I found this below and thought this might be sufficient for some of you.
"OpenBeken features:
*    Tasmota-like setup, configuration and experience on all supported platforms (supports common Tasmota JSON over http and MQTT, etc)"
